### PR TITLE
Update to LSP4J 0.14.0

### LINKED ISSUE #135: Update to lsp4j 0.14.0
Update to lsp4j 0.14.0
https://github.com/eclipse/lsp4j/issues/628
https://github.com/eclipse/lsp4j/pull/630

### PR DESCRIPTION
**Pull request by cdietrich**
Update to LSP4J 0.14.0
Fixes #135

### CODE REVIEW

**Comment by cdietrich** [comment]
release is not out yet

File: target-platforms/target-platform-latest/target-platform-latest.target [hunk 1]
```diff
@@ -13,7 +13,7 @@
 			<unit id="org.eclipse.lsp4j.jsonrpc" version="0.0.0"/>
 			<unit id="org.eclipse.lsp4j.debug" version="0.0.0"/>
 			<unit id="org.eclipse.lsp4j.jsonrpc.debug" version="0.0.0"/>
-			<repository location="https://download.eclipse.org/lsp4j/updates/releases/0.13.0/"/>
+			<repository location="https://download.eclipse.org/lsp4j/updates/releases/0.14.0/"/>
```
Comment (jonahgraham) [inline target-platforms/target-platform-latest/target-platform-latest.target:16]: Release is out now.

**Comment by jonahgraham** [comment]
Error extracted from the PR build
```
[INFO] org.eclipse.lsp4e 0.13.11-SNAPSHOT ................. FAILURE [ 21.237 s]

-- snip --

[ERROR] Failed to execute goal org.eclipse.tycho.extras:tycho-p2-extras-plugin:2.7.0:compare-version-with-baselines (compare-version-with-baseline) on project org.eclipse.lsp4e: Only qualifier changed for (org.eclipse.lsp4e/0.13.11.202205261506). Expected to have bigger x.y.z than what is available in baseline (0.13.11.202205241336) -> [Help 1]
```

**Comment by cdietrich** [comment]
i have no clue how baselines are managed. what needs to be changed?

**Comment by jonahgraham** [comment]
> i have no clue how baselines are managed. what needs to be changed? readme says 1 change per release. was lsp4e already released for 2022-06?

Yes, already released multiple times, including yesterday: https://github.com/eclipse/lsp4e/releases/tag/0.20.4

Your latest patchset builds ok, but tests fail. I assume that they are simply flaky tests, but someone else needs to review.

**Review by jonahgraham — APPROVED** [review]

**Comment by mickaelistria** [comment]
All tests passed in a rebuild; let's merge.
Thanks!

**Pull request merged**